sdk/iot/Azure.Iot.Hub.Service/src/IoTHubServiceClient.cs

```diff
@@ -51,101 +51,80 @@ public class IoTHubServiceClient
-
-            Statistics = new StatisticsClient();
-            Messages = new CloudToDeviceMessagesClient();
-            Files = new FilesClient();
-            Jobs = new JobsClient();
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
-        /// </summary>
-        /// <param name="endpoint">
-        /// The IoT Hub service instance URI to connect to.
-        /// </param>
-        /// <param name="credential">
-        /// The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token.
-        /// </param>
-        public IoTHubServiceClient(Uri endpoint, TokenCredential credential)
-            : this(endpoint, credential, new IoTHubServiceClientOptions())
+        /// <seealso cref="IotHubServiceClient(Uri, IotHubSasCredential, IotHubServiceClientOptions)">
+        /// This other constructor provides an opportunity to override default behavior, including setting the sas token time to live, specifying the service API version,
+        /// overriding <see href="https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Pipeline.md">transport</see>,
+        /// enabling <see href="https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Diagnostics.md">diagnostics</see>,
+        /// and controlling <see href="https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/Configuration.md">retry strategy</see>.
+        /// </seealso>
+        public IotHubServiceClient(string connectionString, IotHubServiceClientOptions options)
+            : this(new IotHubSasCredential(connectionString), options)
         {
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="IoTHubServiceClient"/> class.
+        /// Initializes a new instance of the <see cref="IotHubServiceClient"/> class.
         /// </summary>
         /// <param name="endpoint">
-        /// The IoT Hub service instance URI to connect to.
+        /// The IoT Hub service instance endpoint to connect to.
         /// </param>
         /// <param name="credential">
-        /// The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token.
+        /// The IoT Hub credentials, to be used for authenticating against an IoT Hub instance via SAS tokens.
         /// </param>
         /// <param name="options">
-        /// Options that allow configuration of requests sent to the IoT Hub service.
+        /// (optional) Options that allow configuration of requests sent to the IoT Hub service.
         /// </param>
-        public IoTHubServiceClient(Uri endpoint, TokenCredential credential, IoTHubServiceClientOptions options)
+        public IotHubServiceClient(Uri endpoint, IotHubSasCredential credential, IotHubServiceClientOptions options = default)
```

Comment:
We are exposing the constructor in the format below:
```csharp
var credential = new IotHubSasCredential("shared_access_policy", "shared_access_key");
var client = new IotHubServiceClient(new Uri("uri_here"), credential);
```

We could have had the constructors in the following format instead:
```csharp
var credential = new IotHubSasCredential(new Uri("uri_here"), "shared_access_policy", "shared_access_key");
var client = new IotHubServiceClient(credential);
```
but ideally Uri is a separate concept from credential, and the constructor should not force the user to bind them to the same type. This would also not have been in the same format as `TokenCredential` constructors are.
`public IotHubServiceClient(Uri endpoint, TokenCredential credential, IotHubServiceClientOptions options = default) {}`

Internally, however, since the connectionString constructor sets the endpoint to `IotHubSasCredential`, we do bind the `Uri` to the credential type.